source-code/app/src/main/java/org/buildmlearn/toolkit/utilities/SignerThread.java

```diff
@@ -345,7 +344,7 @@ public void attr(String ns, String name, int resourceId, int type, Object val) {
                 name = null;
             }
 
-            if (!Objects.equals(name, oldName) || val != oldVal) {
+            if ((!((name==null&&oldName==null)||(!(name==null&&oldName!=null))||(!(name!=null&&oldName==null))||(name.equals(oldName)))) || val != oldVal) {
```

Comment:
Yes object.equals is added after api level 19. Btw what I mean by last comment is to simply your new statement using Demorgan's law. If u feel difficult to convert it then feel free to simplify it using if else statement.